### ISSUE TITLE
Add C# 9 UT for S1121

### ISSUE THREAD

**Issue by ds1709** [SonarSource/sonar-dotnet#3774]
### Description

False positive S1121 when copy records (C# 9) using `with` keyword.

### Repro steps
Record sample
```
record Foo
{
    public int Bar { get; init; }
}
```
Call
```
var f1 = new Foo();
var f2 = f1 with { Bar = 10 };
```
### Expected behavior

S1121 warning not occures.

* SonarLint 4.27.0.22695
* Visual Studio 16.8.1

**Comment by pavel-mikula-sonarsource** [comment]
Hi @ds1709,

Thank you for reporting this case and congratulations to being the first one reporting issue on C# 9 topic.

I can confirm this issue and we'll document it for now, as C# 9 is not officially supported yet in our analyzer.
